INAV - No OSD Needed - Compile Error

I have a project where I don't need OSD. When I disable all the OSD references in the target files I get an error compiling the code.

Compile error appears in src/main/rx/rx.c
      undefined reference to 'osdConfig_System'

The src/main/rx/rx.c code includes this.
       #include "rx/crsf.h"

The src/main/rx/crssf.c code includes this.
      #include "io/osd.h"

Also  src/main/rx/crssf.c has code using OSD for calculations.

Recommended Solution:
    #ifdef USE_OSD

I'm using the latest code from GitHub.


![Compile Error](https://user-images.githubusercontent.com/98717468/158278472-d22dcb75-4962-4bd7-aa37-2d977d0f0374.png)

--- a/src/main/fc/config.c
+++ b/src/main/fc/config.c
@@ -340,7 +340,9 @@ void processSaveConfigAndNotify(void)
     writeEEPROM();
     readEEPROM();
     beeperConfirmationBeeps(1);
+#ifdef USE_OSD
     osdShowEEPROMSavedNotification();
+#endif
 }
 
 void writeEEPROM(void)
@@ -370,7 +372,9 @@ void ensureEEPROMContainsValidData(void)
  */
 void saveConfigAndNotify(void)
 {
+#ifdef USE_OSD
     osdStartedSaveProcess();
+#endif
     saveState = SAVESTATE_SAVEANDNOTIFY;
 }
 

--- a/src/main/fc/fc_msp.c
+++ b/src/main/fc/fc_msp.c
@@ -3221,8 +3221,12 @@ static bool mspParameterGroupsCommand(sbuf_t *dst, sbuf_t *src)
 #ifdef USE_SIMULATOR
 bool isOSDTypeSupportedBySimulator(void)
 {
+#ifdef USE_OSD
 	displayPort_t *osdDisplayPort = osdGetDisplayPort();
 	return (osdDisplayPort && osdDisplayPort->cols == 30 && (osdDisplayPort->rows == 13 || osdDisplayPort->rows == 16));
+#else
+    return false;
+#endif
 }
 
 void mspWriteSimulatorOSD(sbuf_t *dst)

--- a/src/main/io/displayport_msp_bf_compat.c
+++ b/src/main/io/displayport_msp_bf_compat.c
@@ -17,7 +17,7 @@
 
 #include "platform.h"
 
-#ifdef USE_MSP_DISPLAYPORT
+#if defined(USE_OSD) && defined(USE_MSP_DISPLAYPORT)
 
 #ifndef DISABLE_MSP_BF_COMPAT
 

--- a/src/main/io/displayport_msp_bf_compat.h
+++ b/src/main/io/displayport_msp_bf_compat.h
@@ -22,7 +22,7 @@
 
 #include "platform.h"
 
-#if defined(USE_MSP_DISPLAYPORT) && !defined(DISABLE_MSP_BF_COMPAT)
+#if defined(USE_OSD) && defined(USE_MSP_DISPLAYPORT) && !defined(DISABLE_MSP_BF_COMPAT)
 #include "osd.h"
 uint8_t getBfCharacter(uint8_t ch, uint8_t page);
 #define isBfCompatibleVideoSystem(osdConfigPtr) (osdConfigPtr->video_system == VIDEO_SYSTEM_BFCOMPAT)

--- a/src/main/rx/crsf.c
+++ b/src/main/rx/crsf.c
@@ -242,6 +242,7 @@ STATIC_UNIT_TESTED uint8_t crsfFrameStatus(rxRuntimeConfig_t *rxRuntimeConfig)
             rxLinkStatistics.uplinkTXPower = crsfTxPowerStatesmW[crsftxpowerindex];
             rxLinkStatistics.activeAntenna = linkStats->activeAntenna;
 
+#ifdef USE_OSD
             if (rxLinkStatistics.uplinkLQ > 0) {
                 int16_t uplinkStrength;   // RSSI dBm converted to %
                 uplinkStrength = constrain((100 * sq((osdConfig()->rssi_dbm_max - osdConfig()->rssi_dbm_min)) - (100 * sq((osdConfig()->rssi_dbm_max  - rxLinkStatistics.uplinkRSSI)))) / sq((osdConfig()->rssi_dbm_max - osdConfig()->rssi_dbm_min)),0,100);
@@ -250,10 +251,10 @@ STATIC_UNIT_TESTED uint8_t crsfFrameStatus(rxRuntimeConfig_t *rxRuntimeConfig)
                 else if (rxLinkStatistics.uplinkRSSI < osdConfig()->rssi_dbm_min)
                     uplinkStrength = 0;
                 lqTrackerSet(rxRuntimeConfig->lqTracker, scaleRange(uplinkStrength, 0, 99, 0, RSSI_MAX_VALUE));
-            }
-            else
+            } else {
                 lqTrackerSet(rxRuntimeConfig->lqTracker, 0);
-
+            }
+#endif
             // This is not RC channels frame, update channel value but don't indicate frame completion
             return RX_FRAME_PENDING;
         }

